File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/statistik/StichprobenQuelle.java
package de.maibornwolff.domainchoreograph.examples.statistik;

import de.maibornwolff.domainchoreograph.examples.statistik.domaintypes.Stichprobe;
import de.maibornwolff.domainchoreograph.examples.statistik.domaintypes.StichprobenAdapter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class StichprobenQuelle implements StichprobenAdapter {

    @Override
    public List<Stichprobe> getStichproben() {
        return generateStichprobenPool();
    }

    public static List<Stichprobe> generateStichprobenPool() {
        List<Stichprobe> stichproben = new ArrayList<>();
        stichproben.add(new Stichprobe(Arrays.asList(1, 2, 3, 4, 5)));
        stichproben.add(new Stichprobe(Arrays.asList(5, 9, 4, 3, 1)));
        stichproben.add(new Stichprobe(Arrays.asList(4, 4, 5)));

        return stichproben;
    }

    @Override
    public String toString() {
        return "(StichprobenQuelle)";
    }
}

File: core/src/test/java/de/maibornwolff/domainchoreograph/example/StichprobenQuelle.kt
package de.maibornwolff.domainchoreograph.example

import de.maibornwolff.domainchoreograph.example.domaindefinitions.StichprobenService
import de.maibornwolff.domainchoreograph.example.domaindefinitions.statistik.Stichprobe
import java.util.*

class StichprobenQuelle : StichprobenService {

    override val stichproben: List<Stichprobe>
        get() = generateStichprobenPool()

    override fun toString(): String {
        return "(StichprobenQuelle)"
    }

    companion object {

        fun generateStichprobenPool(): List<Stichprobe> {
            val stichproben = ArrayList<Stichprobe>()
            stichproben.add(Stichprobe(Arrays.asList(1, 2, 3, 4, 5)))
            stichproben.add(Stichprobe(Arrays.asList(5, 9, 4, 3, 1)))
            stichproben.add(Stichprobe(Arrays.asList(2, 4, 6, 8)))

            return stichproben
        }
    }
}
